extension/leaderelector/README.md

```diff
@@ -0,0 +1,34 @@
+# Leader Elector Extension
+
+This extension enables OpenTelemetry components to run in HA mode across a Kubernetes cluster. The component that owns the lease becomes the leader and becomes the active instance.
+## How It Works
+
+The extension uses k8s.io/client-go/tools/leaderelection to perform leader election. The component that owns the lease becomes the leader and runs the function defined in onStartedLeading. If the leader loses the lease, it runs the function defined in onStoppedLeading, stops its operation, and waits to acquire the lease again.
+## Configuration
+
+```yaml
+receivers:
+  my_awesome_receiver:
+    leader_elector: leader_elector
+extensions:
+  leader_elector:
+    auth_type: kubeConfig
+    lease_name: foo
+    lease_namespace: default
+
+service:
+  extensions: [leader_elector]
+  pipelines:
+    metrics:
+      receivers: [my_awesome_receiver]
+```
+
+### Leader Election Configuration
```

Comment:
It would be better to give a rolebingding example  about the `lease` that serviceaccount needs to bind with `ClusterRoleBinding` `ClusterRole`.